### ISSUE TITLE
Regarding the Dimension Issue of mvmap

### ISSUE THREAD

**Issue by yunzz** [marukosan93/RS-rPPG#6]
Hi Marko,

Thank you for sharing your code and preprocessing scripts. I have a question regarding the dimensions of the preprocessed maps:

From the paper and your repository, MSTmaps，BGmaps and TMaps2 are described as having 6 channels (RGB + YUW), and MVmaps are supposed to have the same shape [63, N, 6].

However, in the create_mvmaps_VIPL.sh / example code, the generated MVmap seems to have only 3 channels, representing (x_center, y_center, area) for each ROI combination.

Could you clarify:

Are MVmaps supposed to be 6 channels?

If yes, how are the additional channels derived from the original 3 (x, y, area)?

Thanks a lot for your help!

Best regards,
yunzz

**Comment by marukosan93** [comment]
Hi,
So basically the other maps are all 6 channels as they are either RGB+YUV, or in case of TMAP the 6 channels come from the 6 methods. For the movement estimates, as we have x,y,area these are only 3 channels, but in order to keep the same dimensionality, in the dataloader it is simply duplicated to get 6 channels. 

 mvmap = torch.cat([mvmap,mvmap],dim=0)   (MST_tmap2_mv_bg_aug2_sunet_snr_nosq.py line 278)

**Comment by yunzz** [comment]
Thanks for the clarification! That makes sense now.

Just to double-check:

In the README, it says the BGmap channels are Y, U, W,

But in the code (generate_bg_signal_map()), the six channels are actually R, G, B, Y, U, V (i.e., RGB + YUV).

So to confirm, the correct version should be RGB + YUV (R, G, B, Y, U, V), and the README mentioning YUW is just a typo, right?

**Comment by marukosan93** [comment]
Yes exactly, good catch, thanks. Could you just quickly copy/paste the part of the readme with the typo so I can fix it?

**Comment by yunzz** [comment]
Here’s the README part with the typo fixed (YUW → YUV). You can just copy/paste it directly.

The original videos are firstly preprocessed by extracting the MSTmaps following https://github.com/nxsEdson/CVD-Physiological-Measurement. Both the MSTmaps and groundtruth bvp are resampled to 30 fps. To get the TMap positive augmentation maps use create_tmaps2_VIPL.sh on your dataset. It is similar to the create_tmaps_VIPL.sh from RS-rPPG, but with more traditional methods resulting in 6 channels. In the example code we assume the data used in pre-processed from PURE, OBF, MMSE, VIPL-HR datasets, but can't provide the actual data or preprocessed files. The structure of the data that can be used with our dataloader is:
Dataset1:
├── Sample1 (signal maps and ground truth are long N frames which is the total length of each video)
├──├── mstmap.npy (contains [63,N,6] multi-scale spatial-temporal maps (RGB+YUV) calculated from videos)
├──├── tmap2.npy (contains [63,N,6] traditional augmenation calculated from the mstmaps by using create_tmaps2_VIPL.sh)
├──├── bvp.npy ( array with grountruth bvp signal [N])
├──├── mvmap.npy (contains [63,N,6] movement maps, calculated from the ROI locations consistent with MSTmap signal row, obtained using create_mvmaps_VIPL.sh)
├──├── bgmap.npy (contains [63,N,6] background maps, calculated from 6 background regions in an analogous way as MSTmaps, obtained using create_bgmaps_VIPL.sh)


If your dataset is processed in this way, with some minimal code changes you can use our dataloader as described by the following steps. The preprocessing has a couple of extra steps compared to RS-rPPG, tmaps are now with 6 traditional methods, mstmaps also have 6 channels (RGB+YUV), BGmaps and MVmaps are calculated during preprocessing. For FRmaps the augmentations are created by the dataloader, as they frequencies are randomly sampled.

**Comment by marukosan93** [comment]
Got it now, thanks a lot!

**Comment by yunzz** [comment]
Great! I’ll continue studying then — thanks again for your work, it really inspires me and makes me more interested in doing research.
